src/mngt/ocf_mngt_flush.c

```diff
@@ -141,56 +141,61 @@ static int _ocf_mngt_get_sectors(ocf_cache_t cache, ocf_core_id_t core_id,
 	uint64_t core_line;
 	ocf_core_id_t i_core_id;
 	struct flush_data *p;
-	uint32_t i, j, dirty = 0;
+	uint32_t i, dirty_found = 0, dirty_total = 0;
+	unsigned ret = 0;
 
-	dirty = env_atomic_read(&core->runtime_meta->dirty_clines);
-	if (!dirty) {
+	ocf_metadata_start_exclusive_access(&cache->metadata.lock);
+
+	dirty_total = env_atomic_read(&core->runtime_meta->dirty_clines);
+	if (!dirty_total) {
 		*num = 0;
 		*tbl = NULL;
-		return 0;
+		goto unlock;
 	}
 
-	p = env_vmalloc(dirty * sizeof(**tbl));
-	if (!p)
-		return -OCF_ERR_NO_MEM;
+	p = env_vmalloc(dirty_total * sizeof(**tbl));
+	if (!p) {
+		ret = -OCF_ERR_NO_MEM;
+		goto unlock;
+	}
 
-	for (i = 0, j = 0; i < cache->device->collision_table_entries; i++) {
+	for (i = 0; i < cache->device->collision_table_entries; i++) {
 		ocf_metadata_get_core_info(cache, i, &i_core_id, &core_line);
 
-		if (i_core_id != core_id)
-			continue;
-
-		if (!metadata_test_valid_any(cache, i))
-			continue;
-
-		if (!metadata_test_dirty(cache, i))
-			continue;
-
-		if (ocf_cache_line_is_used(cache, i))
-			continue;
+		if (i_core_id == core_id &&
+				metadata_test_valid_any(cache, i) &&
+				metadata_test_dirty(cache, i)) {
```

Comment:
How about warping it within macro or inline function `cache_line_is_dirty`?

Reply:
I would rather not, now these functions have a single purpose and the names are descriptive. cache_line_is_dirty would easily be confused with cacheline_test_dirty .